agent/tests/test_lib_aws_sdk_php.c

```diff
@@ -458,6 +482,103 @@ static void test_nr_lib_aws_sdk_php_handle_version(void) {
   tlib_php_request_end();
 }
 
+#if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO
+static void test_nr_lib_aws_sdk_php_lambda_invoke() {
+  tlib_php_engine_create("newrelic.cloud.aws.account_id=\"test_account\"");
+  tlib_php_request_start();
+
+  tlib_php_request_eval("function lambda_invoke($a, $b) { return; }");
+  nr_php_wrap_user_function(NR_PSTR("lambda_invoke"), aws_lambda_invoke_wrapper);
+
+  /* Test full-info run */
+  char* args
+      = "array("
+        "    0 => array("
+        "        'FunctionName' => 'us-east-2:012345678901:function:my-function'"
+        "    )"
+        ")";
+  zval* array_arg = tlib_php_request_eval_expr(args);
+  char* expect = "'arn:aws:lambda:us-east-2:012345678901:function:my-function'";
+  zval* expect_arg = tlib_php_request_eval_expr(expect);
+  zval* expr = nr_php_call(NULL, "lambda_invoke", expect_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&expect_arg);
+  nr_php_zval_free(&array_arg);
+
+  /* Test alias full-info run */
+  args
+      = "array("
+        "    0 => array("
+        "        'FunctionName' => 'us-east-2:012345678901:function:my-function:v1'"
+        "    )"
+        ")";
+  array_arg = tlib_php_request_eval_expr(args);
+  expect = "'arn:aws:lambda:us-east-2:012345678901:function:my-function:v1'";
+  expect_arg = tlib_php_request_eval_expr(expect);
+  expr = nr_php_call(NULL, "lambda_invoke", expect_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&expect_arg);
+  nr_php_zval_free(&array_arg);
+
+  /* Test INI extract */
+  args
+      = "array("
+        "    0 => array("
+        "        'FunctionName' => 'us-east-2:my-function'"
+        "    )"
+        ")";
+  array_arg = tlib_php_request_eval_expr(args);
+  expect = "'arn:aws:lambda:us-east-2:test_account:function:my-function'";
+  expect_arg = tlib_php_request_eval_expr(expect);
+  expr = nr_php_call(NULL, "lambda_invoke", expect_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&expect_arg);
+  nr_php_zval_free(&array_arg);
+
+  /* Test invalid arg 1 */
+  args
+      = "array("
+        "    0 => array("
+        "        'FunctionName' => 123"
```

Comment:
Can you please add a test for where functionName is an empty string and where it is null?

Reply:
added in [5c45a0e](https://github.com/newrelic/newrelic-php-agent/pull/1023/commits/5c45a0ef346dbb2bf7e1b28a8ed0de4cbabcbb3f)